src/Umbraco.Web/Trees/ApplicationTreeController.cs

```diff
@@ -71,9 +86,64 @@ public async Task<SectionRootNode> GetApplicationTrees(string application, strin
                 }
             }
 
-            var multiTree = SectionRootNode.CreateMultiTreeSectionRoot(rootId, collection);
-            multiTree.Name = Services.TextService.Localize("sections/"+ application);
-            return multiTree;
+            //Don't apply fancy grouping logic futher down, if we are not 'settings' section
+            if(application != Constants.Applications.Settings)
+            {
+                var multiTree = SectionRootNode.CreateMultiTreeSectionRoot(rootId, collection);
+                multiTree.Name = Services.TextService.Localize("sections/" + application);
+
+                rootNodeGroups.Add(multiTree);
+                return rootNodeGroups;
+            }
+
+            //For settings section only
+            //Group trees by [CoreTree] attribute
+
+            //Core Trees contains all trees for all sections/applications
+            foreach(var treeSectionGroup in CoreTrees.Value)
+            {
+                var treeGroupName = treeSectionGroup.Key;
+
+                var groupNodeCollection = new TreeNodeCollection();
+
+                //Only add trees to a new collection if they are from 'settings'
+                foreach (var treeItem in treeSectionGroup)
+                {
+                    //Item1 tuple - is the type from all tree types
+                    var treeItemType = treeItem.Item1;
+
+                    var findAppTree = appTrees.SingleOrDefault(x => x.GetRuntimeType() == treeItemType);
+                    if (findAppTree != null)
+                    {
+                        //Now we need to get the 'TreeNode' which is in 'collection'
+                        var treeItemNode = collection.SingleOrDefault(x => x.AdditionalData["treeAlias"].ToString() == findAppTree.Alias);
+
+                        if (treeItemNode != null)
+                        {
+                            //Add to a new list/collection
+                            groupNodeCollection.Add(treeItemNode);
+                        }
+                    }
+                }
+
+                //If treeGroupName == null then its third party
+                if (treeGroupName == null)
+                {
+                    //This is used for the localisation key
+                    //treeHeaders/thirdPartyGroup
+                    treeGroupName = "thirdPartyGroup";
+                }
+
+                if (groupNodeCollection.Any())
```

Comment:
`groupNodeCollection` is a List<TreeNode>, since you are using a List you should be using `groupNodeCollection.Count > 0`. Try not to use `Any()` or Linq stuff when you don't need to to avoid more iterations/allocations